Xamarin.Essentials/Battery/Battery.shared.cs

```diff
@@ -38,29 +29,14 @@ public static partial class Battery
             }
         }
 
-        static void SetCurrent()
-        {
-            currentLevel = Battery.ChargeLevel;
-            currentSource = Battery.PowerSource;
-            currentState = Battery.State;
-        }
-
         static void OnBatteryChanged(double level, BatteryState state, BatteryPowerSource source)
             => OnBatteryChanged(new BatteryChangedEventArgs(level, state, source));
 
         static void OnBatteryChanged()
             => OnBatteryChanged(ChargeLevel, State, PowerSource);
 
         static void OnBatteryChanged(BatteryChangedEventArgs e)
-        {
-            if (currentLevel != e.ChargeLevel ||
-                currentSource != e.PowerSource ||
-                currentState != e.State)
-            {
-                SetCurrent();
-                BatteryChanagedInternal?.Invoke(e);
-            }
-        }
+            => BatteryChanagedInternal?.Invoke(e);
```

Comment:
Any reason to remove these checks? Else the user will get multiple notifications when things change and would be very annoying.

Reply:
Does the OS report a change, but nothing has actually changed?

Reply:
it is highly possible. Sometimes things trigger back and forth. This was a big complaint with my connectivity plugin that it didn't do enough checks. We can test though I guess..... Maybe put a counter on the sample page?